src/main/java/io/confluent/connect/hdfs/json/JsonFormat.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.confluent.connect.hdfs.json;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.kafka.connect.json.JsonConverter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
+import io.confluent.connect.hdfs.storage.HdfsStorage;
+import io.confluent.connect.storage.format.Format;
+import io.confluent.connect.storage.format.RecordWriterProvider;
+import io.confluent.connect.storage.format.SchemaFileReader;
+import io.confluent.connect.storage.hive.HiveFactory;
+
+public class JsonFormat implements Format<HdfsSinkConnectorConfig, Path> {
```

Comment:
IIUC, this is the class that users have to specify within the connector's configuration. Wouldn't it be better to have JavaDoc on this class with explanations about how to use it?

Reply:
I agree with the necessity of javadocs here and in similar places. 

However when documentation is missing from the older/prexisting similar classes as well, I'm inclined to treat it as tech debt and address it altogether in a separate PR. I find this a reasonable compromise that aims to avoid introducing too much noise in focused PRs and helps not duplicating the effort (write the javadocs once for everything). Of course, there's always the danger to postpone this task a bit too much, but I believe that with the appropriate jiras it could be tracked sufficiently.

Reply:
Wouldn't it be better to not incur more tech debt, especially when the change is fresh in your mind? Seems cheap to do now for this class and expensive to do later.

Reply:
I was referring to the absence of javadoc in `AvroFormat` and `ParquetFormat`. I don't mind adding javadoc in `JsonFormat` but I don't think the irrelevant files should be touched as part of this PR.

Reply:
I agree with that: add it here, but not anywhere else.

Reply:
I'm a big fan of javadocs. Added a few class-level ones. You might have noticed the wide absence of javadocs in our connectors (storage-common has some). Given that, their addition needs to follow a more systematic approach.